controllers/glance_controller.go

```diff
@@ -608,103 +618,125 @@ func (r *GlanceReconciler) reconcileNormal(ctx context.Context, instance *glance
-// the deployment logic itself, as well as the output settings mirrored in the
-// main Glance CR status
-func (r *GlanceReconciler) apiDeployment(ctx context.Context, instance *glancev1.Glance, current glancev1.GlanceAPITemplate, helper *helper.Helper) error {
-
-	// By default internal and external points to diff instances, but we might
-	// want to override "external" with "single" in case APIType == "single":
-	// in this case we only deploy the External instance and skip the internal
-	// one
-	var internal string = glancev1.APIInternal
-	var external string = glancev1.APIExternal
-
-	// If we're deploying a "single" instance, we skip GlanceAPI.Internal, and
-	// we only deploy the External instance passing "glancev1.APISingle" to the
-	// GlanceAPI controller, so we can properly handle this use case (nad and
-	// service creation).
-	if current.Type == "single" {
-		external = glancev1.APISingle
-	}
-	glanceAPI, op, err := r.apiDeploymentCreateOrUpdate(ctx, instance, instance.Spec.GlanceAPI, external, helper)
+// in the main CR according to a given strategy (split vs single). The main
+// switch-case is usefult to define a clear logic associated with the defined
+// layout
+func (r *GlanceReconciler) apiDeployment(ctx context.Context, instance *glancev1.Glance, instanceName string, current glancev1.GlanceAPITemplate, helper *helper.Helper) error {
+
+	var glanceAPI *glancev1.GlanceAPI
+	var err error
+	switch current.Type {
+	case "single":
+		// Deploy a single instance of GlanceAPI
+		glanceAPI, err = r.ensureAPIDeployment(ctx, instance, glancev1.APISingle, instanceName, helper)
+		if err != nil {
+			return err
+		}
+	case "split":
+		// Deploy a layout composed by an internal and external instance
+		// TODO: (fpantano) If the backend is not set prevent this layout deployment
+		_, err = r.ensureAPIDeployment(ctx, instance, glancev1.APIExternal, instanceName, helper)
+		if err != nil {
+			return err
+		}
+		glanceAPI, err = r.ensureAPIDeployment(ctx, instance, glancev1.APIInternal, instanceName, helper)
+		if err != nil {
+			return err
+		}
+	default:
+		// by default we split and deploy a layout composed by an internal and an external instance
+		// TODO: (fpantano) If the backend is not set prevent this layout deployment
+		_, err = r.ensureAPIDeployment(ctx, instance, glancev1.APIExternal, instanceName, helper)
+		if err != nil {
+			return err
+		}
+		glanceAPI, err = r.ensureAPIDeployment(ctx, instance, glancev1.APIInternal, instanceName, helper)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Get external GlanceAPI's condition status and compare it against priority of internal GlanceAPI's condition
+	apiCondition := glanceAPI.Status.Conditions.Mirror(glancev1.GlanceAPIReadyCondition)
+	// Get internal GlanceAPI's condition status for comparison with external below
+	internalAPICondition := glanceAPI.Status.Conditions.Mirror(glancev1.GlanceAPIReadyCondition)
```

Comment:
Yeah, I'm not really doing a real distinction in the `GlanceAPIReadyCondition` for two reasons:
1. it applies well to `single` instance deployment;
2. both internal and external logically represent an API that serves a given workload

Do you think I should introduce a:

1. `GlanceAPIInternalReadyCondition`
2. `GlanceAPIExternalReadyCondition`
 
and see if it makes sense growing this more for `edge` and `single` deployments?
This might be content for a follow up patch, @abays wdyt?